tests/bdd/features/volume/resize/test_resize_offline.py

```diff
@@ -0,0 +1,306 @@
+# It starts the deployer which launches all the necessary containers.
+# Pools are created for convenience such that they are available for use by the tests.
+@pytest.fixture(autouse=True, scope="module")
+def init(disks):
+    Deployer.start(3)
+    assert len(disks) == DEFAULT_REPLICA_CNT
+    ApiClient.pools_api().put_node_pool(
+        NODE1_NAME, POOL1_UUID, CreatePoolBody([f"aio://{disks[0]}"])
+    )
+    ApiClient.pools_api().put_node_pool(
+        NODE2_NAME, POOL2_UUID, CreatePoolBody([f"aio://{disks[1]}"])
+    )
+    ApiClient.pools_api().put_node_pool(
+        NODE3_NAME, POOL3_UUID, CreatePoolBody([f"aio://{disks[2]}"])
+    )
+    pytest.replica_count = DEFAULT_REPLICA_CNT
+    yield
+    Deployer.stop()
+
+
+@pytest.fixture(scope="function")
+def create_volume():
+    volume = ApiClient.volumes_api().put_volume(
+        VOLUME_UUID,
+        CreateVolumeBody(
+            VolumePolicy(False), int(pytest.replica_count), VOLUME_SIZE, False
+        ),
+    )
+    assert volume.spec.uuid == VOLUME_UUID
+    replicas = volume.state.replica_topology
+    assert len(replicas) == int(pytest.replica_count)
+    yield
+    del_volume()
+
+
+# A jumper function so that we can set replica count in cases that need
+# to run with different replica counts.
+def volume(create_volume):
+    """Invoke volume creation"""
+
+
+@scenario("resize_offline.feature", "expand a published volume")
+def test_expand_a_published_volume():
+    """expand a published volume."""
+
+
+@scenario("resize_offline.feature", "expand an unpublished volume")
+def test_expand_an_unpublished_volume():
+    """expand an unpublished volume."""
+
+
+@scenario(
+    "resize_offline.feature", "expand an unpublished volume and then make it published"
+)
+def test_expand_an_unpublished_volume_and_then_make_it_published():
+    """expand an unpublished volume and then make it published."""
+
+
+@scenario(
+    "resize_offline.feature", "expand an unpublished volume having a not-online replica"
+)
+def test_expand_an_unpublished_volume_having_a_notonline_replica():
+    """expand an unpublished volume having a not-online replica."""
+
+
+@scenario("resize_offline.feature", "shrink an unpublished volume")
+def test_shrink_an_unpublished_volume():
+    """shrink an unpublished volume."""
+
+
+@given("a published volume with more than one replica and all are healthy")
+def a_published_volume_with_more_than_one_replica_and_all_are_healthy(create_volume):
+    """a published volume with more than one replica and all are healthy."""
+    the_volume_is_published()
+    pytest.exception = None
+
+
+@given(
+    parsers.parse(
+        "an unpublished volume with {repl_count:d} replicas and all are healthy"
+    )
+)
+def an_unpublished_volume_with_repl_count_replicas_and_all_are_healthy(repl_count):
+    """an unpublished volume with <repl_count> replicas and all are healthy."""
+    pytest.replica_count = repl_count
+    volume(create_volume)
```

Comment:
I thought this was working bcz tests passed but this isn't correct. It worked because expand step has `create_volume` fixture.
@tiagolobocastro @Abhinandan-Purkait Looking for a way to take variable `repl_count` and pass it to fixture `create_volume`, any suggestions?

Reply:
There is indeed a way, let me try that and let you know.
`@pytest.mark.parametrize('tester_arg', [True, False])` using this thing.